pkg/activator/testing/roundtripper.go

```diff
@@ -137,6 +153,17 @@ func (rt *FakeRoundTripper) RT(req *http.Request) (*http.Response, error) {
 		resp = defaultRequestResponse()
 	}
 
+	// Delay if set before sending response
+	if resp.Delay.Seconds() != 0 {
```

Comment:
Instead of repeating this here should we just dedupe this code and put it at the beginning of the `RT` method?

Reply:
The could could be defined as a function ahead of the call, but it needs to be called after the `resp` value is fetched which happens in two different code paths.

Reply:
> could be defined as a function ahead of the call

let's do that

Reply:
This is updated now.